installer/bundle/bundle_skel_AIX.sh

```diff
@@ -0,0 +1,432 @@
+    echo "  --purge                Uninstall the package and remove all related data."
+    echo "  --remove               Uninstall the package from the system."
+    echo "  --restart-deps         Reconfigure and restart dependent service"
+    echo "  --source-references    Show source code reference hashes."
+    echo "  --upgrade              Upgrade the package in the system."
+    echo "  --version              Version of this shell bundle."
+    echo "  --debug                use shell debug mode."
+    echo "  -? | --help            shows this usage text."
+}
+
+source_references()
+{
+    cat <<EOF
+-- Source code references --
+EOF
+}
+
+cleanup_and_exit()
+{
+    # $1: Exit status
+    # $2: Non-blank (if we're not to delete bundles), otherwise empty
+
+    if [ -z "$2" -a -d "$EXTRACT_DIR" ]; then
+        cd $EXTRACT_DIR/..
+        rm -rf $EXTRACT_DIR
+    fi
+
+    if [ -n "$1" ]; then
+        exit $1
+    else
+        exit 0
+    fi
+}
+
+getVersionNumber()
+{
+    # Parse a version number from a string.
+    #
+    # Parameter 1: string to parse version number string from
+    #     (should contain something like mumble-4.2.2.135.universal.x86.tar)
+    # Parameter 2: prefix to remove ("mumble-" in above example)
+
+    if [ $# -ne 2 ]; then
+        echo "INTERNAL ERROR: Incorrect number of parameters passed to getVersionNumber" >&2
+        cleanup_and_exit 1
+    fi
+
+    echo $1 | sed -e "s/$2//" -e 's/\.universal\..*//' -e 's/\.x64.*//' -e 's/\.x86.*//' -e 's/-/./'
+}
+
+verifyNoInstallationOption()
+{
+    if [ -n "${installMode}" ]; then
+        echo "$0: Conflicting qualifiers, exiting" >&2
+        cleanup_and_exit 1
+    fi
+
+    return;
+}
+
+# $1 - The name of the package to check as to whether it's installed
+check_if_pkg_is_installed() {
+    lslpp $1 2> /dev/null 1> /dev/null
+    return $?
+}
+
+# $1 - The filename of the package to be installed
+# $2 - The package name of the package to be installed
+pkg_add() {
+    pkg_filename=$1
+    pkg_name=$2
+
+    echo "----- Installing package: $pkg_name ($pkg_filename) -----"
+    /usr/sbin/installp -a -X -d $pkg_filename ${pkg_name}.rte
+}
+
+# $1 - The package name of the package to be uninstalled
+# $2 - Optional parameter. Only used when forcibly removing omi on SunOS
+pkg_rm() {
+    echo "----- Removing package: $1 -----"
+    /usr/sbin/installp -u $1.rte # 1> /dev/null 2> /dev/null
+}
+
+
+# $1 - The filename of the package to be installed
+# $2 - The package name of the package to be installed
+# $3 - Okay to upgrade the package? (Optional)
+pkg_upd() {
+    pkg_filename=$1
+    pkg_name=$2
+    pkg_allowed=$3
+
+    echo "----- Updating package: $pkg_name ($pkg_filename) -----"
+
+    if [ -z "${forceFlag}" -a -n "$pkg_allowed" ]; then
+        if [ $pkg_allowed -ne 0 ]; then
+            echo "Skipping package since existing version >= version available"
+            return 0
+        fi
+    fi
```

Comment:
The notion of "skipping packages" is all based around version detection for Azure, which isn't appropriate for non-Linux platforms. Please verify how this is called and, if so, remove this 'if' condition. Makes pkg_upd about as simple as pkg_add.